### PR TITLE
Upstream a variety of forms tests from jsdom

### PR DESCRIPTION
**Pull request by domenic**
This upstreams most tests from https://github.com/tmpvar/jsdom/tree/d078830eafd17ec38aa816cf283da39ec5316e31/test/web-platform-tests/to-upstream/html/semantics/forms. I omitted ones that seem to fail in all browsers (spec or jsdom bug, most likely) or ones related to text selection (as the spec there has been changing recently and I didn't want to re-vet the tests). I also fixed one of the tests that was failing in all browsers per https://github.com/whatwg/html/pull/2662.

This still reveals a lot of cases where browsers are incompatible, so I am pretty happy with these tests.

Most of these have been reviewed upstream, but more review would always be appreciated.

/cc @tkent-google

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/230936330)

# Firefox (nightly) #
Testing web-platform-tests at revision 6548cb09c20ed6696dc11c5cd2757563094f48a4
Using browser at version BuildID 20170510183715; SourceStamp ce2218406119c36a551e3faea4e192186ee46cc5
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>13 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/resetting-a-form/reset-form-2.html">/html/semantics/forms/resetting-a-form/reset-form-2.html</a></summary>


|                          Subtest                           | Results |                                       Messages                                      |
|------------------------------------------------------------|---------|-------------------------------------------------------------------------------------|
|                                                            | OK      |                                                                                     |
| `integration test on reset for a created-from-script form` | FAIL    | `assert_equals: option should reset dirtyness to false expected false but got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-button-element/button-click-submits.html">/html/semantics/forms/the-button-element/button-click-submits.html</a></summary>


|                                      Subtest                                      | Results |     Messages     |
|-----------------------------------------------------------------------------------|---------|------------------|
|                                                                                   | TIMEOUT |                  |
| `clicking a button with .click() should trigger a submit`                         | TIMEOUT | `Test timed out` |
| `clicking a button by dispatching an event should trigger a submit`               | PASS    |                  |
| `clicking a button that prevents the event's default should not trigger a submit` | PASS    |                  |
| `clicking a disabled button (via disabled attribute) should not trigger submit`   | PASS    |                  |
| `clicking a disabled button (via ancestor fieldset) should not trigger submit`    | PASS    |                  |
| `clicking a button inside a disabled fieldset's legend *should* trigger submit`   | PASS    |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-button-element/button-type.html">/html/semantics/forms/the-button-element/button-type.html</a></summary>


|                            Subtest                             | Results |                      Messages                     |
|----------------------------------------------------------------|---------|---------------------------------------------------|
|                                                                | OK      |                                                   |
| `a button's type should be submit by default`                  | PASS    |                                                   |
| `a button's type should stay within the range of valid values` | FAIL    | `assert_equals: expected "menu" but got "submit"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-form-element/form-action.html">/html/semantics/forms/the-form-element/form-action.html</a></summary>


|                                                Subtest                                                 | Results |                                                           Messages                                                          |
|--------------------------------------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------------------|
|                                                                                                        | OK      |                                                                                                                             |
| `An action URL should be resolved relative to the document's URL`                                      | PASS    |                                                                                                                             |
| `An empty-string action content attribute should cause the IDL attribute to return the document's URL` | FAIL    | `assert_equals: expected "http://web-platform.test:8000/html/semantics/forms/the-form-element/form-action.html" but got ""` |
| `A missing action content attribute should cause the IDL attribute to return the empty string`         | PASS    |                                                                                                                             |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-input-element/checkbox-click-events.html">/html/semantics/forms/the-input-element/checkbox-click-events.html</a></summary>


|                                                       Subtest                                                       | Results |                            Messages                            |
|---------------------------------------------------------------------------------------------------------------------|---------|----------------------------------------------------------------|
|                                                                                                                     | OK      |                                                                |
| `clicking and preventDefaulting a checkbox causes the checkbox to be checked during the click handler but reverted` | FAIL    | `assert_array_equals: property 0, expected true but got false` |
| `a checkbox input emits click, input, change events in order after synthetic click`                                 | PASS    |                                                                |
| `a checkbox input emits click, input, change events in order after dispatching click event`                         | PASS    |                                                                |
| `checkbox input respects cancel behavior on synthetic clicks`                                                       | PASS    |                                                                |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-input-element/radio-input-cancel.html">/html/semantics/forms/the-input-element/radio-input-cancel.html</a></summary>


|                   Subtest                   | Results | Messages |
|---------------------------------------------|---------|----------|
|                                             | OK      |          |
| `radio input cancel behavior reverts state` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-label-element/proxy-click-to-associated-element.html">/html/semantics/forms/the-label-element/proxy-click-to-associated-element.html</a></summary>


|                                       Subtest                                        | Results | Messages |
|--------------------------------------------------------------------------------------|---------|----------|
|                                                                                      | OK      |          |
| `label with for attribute should proxy click events to the associated element`       | PASS    |          |
| `label without for attribute should proxy click events to the first labelable child` | PASS    |          |
| `clicking a label that prevents the event's default should not proxy click events`   | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-ask-for-a-reset.html">/html/semantics/forms/the-option-element/option-ask-for-a-reset.html</a></summary>


|             Subtest              | Results |                                                       Messages                                                      |
|----------------------------------|---------|---------------------------------------------------------------------------------------------------------------------|
|                                  | OK      |                                                                                                                     |
| `option element ask-for-a-reset` | FAIL    | `uncaught exception: Error: assert_equals: Setting defaultSelected should reset others expected true but got false` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-element-constructor.html">/html/semantics/forms/the-option-element/option-element-constructor.html</a></summary>


|                                                                Subtest                                                                 | Results |                 Messages                |
|----------------------------------------------------------------------------------------------------------------------------------------|---------|-----------------------------------------|
|                                                                                                                                        | OK      |                                         |
| `Option constructor with no arguments`                                                                                                 | PASS    |                                         |
| `Option constructor with falsy arguments`                                                                                              | PASS    |                                         |
| `Option constructor creates HTMLOptionElement with specified text and value`                                                           | PASS    |                                         |
| `Option constructor handles selectedness correctly when specified with defaultSelected only`                                           | FAIL    | `assert_false: expected false got true` |
| `Option constructor handles selectedness correctly, even when incongruous with defaultSelected`                                        | PASS    |                                         |
| `Option constructor treats undefined text and value correctly`                                                                         | PASS    |                                         |
| `Option constructor treats falsy selected and defaultSelected correctly`                                                               | PASS    |                                         |
| `Option constructor treats truthy selected and defaultSelected correctly`                                                              | PASS    |                                         |
| `Option constructor does not set dirtiness (so, manipulating the selected content attribute still updates the selected IDL attribute)` | FAIL    | `assert_false: expected false got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-index.html">/html/semantics/forms/the-option-element/option-index.html</a></summary>


|                                Subtest                                 | Results | Messages |
|------------------------------------------------------------------------|---------|----------|
|                                                                        | OK      |          |
| `option index should work inside the document`                         | PASS    |          |
| `option index should always be 0 for options in datalists`             | PASS    |          |
| `option index should always be 0 for options with no container`        | PASS    |          |
| `option index should always be 0 for options not even in the document` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-select-element/select-multiple.html">/html/semantics/forms/the-select-element/select-multiple.html</a></summary>


|                         Subtest                         | Results | Messages |
|---------------------------------------------------------|---------|----------|
|                                                         | OK      |          |
| `multiple selected options exist, both set from markup` | PASS    |          |
| `multiple selected options exist, one set from script`  | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-select-element/select-selectedOptions.html">/html/semantics/forms/the-select-element/select-selectedOptions.html</a></summary>


|                                                     Subtest                                                      | Results | Messages |
|------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                  | OK      |          |
| `.selectedOptions with no selected option`                                                                       | PASS    |          |
| `.selectedOptions with one selected option`                                                                      | PASS    |          |
| `.selectedOptions using the 'multiple' attribute with no selected options`                                       | PASS    |          |
| `.selectedOptions using the 'multiple' attribute with two selected options`                                      | PASS    |          |
| `.selectedOptions without the 'multiple' attribute but more than one selected option should return the last one` | PASS    |          |
| `.selectedOptions should always return the same value - [SameObject]`                                            | PASS    |          |
| `.selectedOptions should return the same object after selection changes - [SameObject]`                          | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html</a></summary>


|                                            Subtest                                             | Results |                                                               Messages                                                              |
|------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                | OK      |                                                                                                                                     |
| `defaultValue and value are the empty string by default`                                       | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by setting textContent`                                   | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by textContent in combination with appending a text node` | PASS    |                                                                                                                                     |
| `defaultValue and value treat CRLF differently`                                                | FAIL    | `assert_equals: The value property should normalize CRLF and CR to LF expected "foo\nbar\nbaz\nqux" but got "foo\r\nbar\rbaz\nqux"` |
| `tests for the value setter`                                                                   | PASS    |                                                                                                                                     |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/230936336)

# Chrome (unstable) #
Testing web-platform-tests at revision 6548cb09c20ed6696dc11c5cd2757563094f48a4

File: html/semantics/forms/the-button-element/button-type.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const button = document.createElement("button");
+  assert_equals(button.type, "submit");
+
+}, "a button's type should be submit by default");
+
+test(() => {
+
+  const button = document.createElement("button");
+
+  for (const type of ["reset", "button", "menu", "submit"]) {
```
Comment (tkent-google) [inline html/semantics/forms/the-button-element/button-type.html]: WHATWG HTML doesn't have type=menu.

https://html.spec.whatwg.org/multipage/forms.html#attr-button-type

Comment (domenic) [inline html/semantics/forms/the-button-element/button-type.html]: Thanks, fixed!

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/232599169)

# Sauce (safari) #
Testing web-platform-tests at revision 4dda02708fdfd837406b612708668c22b3c6d0b3
Using browser at version 10.0
Starting 10 test iterations
## Unstable results ##


|                                 Test                                 |                                      Subtest                                      |             Results              |     Messages     |
|----------------------------------------------------------------------|-----------------------------------------------------------------------------------|----------------------------------|------------------|
| `/html/semantics/forms/the-button-element/button-click-submits.html` | `clicking a button with .click() should trigger a submit`                         | **PASS: 5/10, MISSING: 5/10**    |                  |
| `/html/semantics/forms/the-button-element/button-click-submits.html` | `clicking a button by dispatching an event should trigger a submit`               | **PASS: 5/10, MISSING: 5/10**    |                  |
| `/html/semantics/forms/the-button-element/button-click-submits.html` | `clicking a button that prevents the event's default should not trigger a submit` | **TIMEOUT: 5/10, MISSING: 5/10** | `Test timed out` |
| `/html/semantics/forms/the-button-element/button-click-submits.html` | `clicking a disabled button (via disabled attribute) should not trigger submit`   | **TIMEOUT: 5/10, MISSING: 5/10** | `Test timed out` |
| `/html/semantics/forms/the-button-element/button-click-submits.html` | `clicking a disabled button (via ancestor fieldset) should not trigger submit`    | **TIMEOUT: 5/10, MISSING: 5/10** | `Test timed out` |
| `/html/semantics/forms/the-button-element/button-click-submits.html` | `clicking a button inside a disabled fieldset's legend *should* trigger submit`   | **PASS: 5/10, MISSING: 5/10**    |                  |

## All results ##
<details>
<summary>12 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/resetting-a-form/reset-form-2.html">/html/semantics/forms/resetting-a-form/reset-form-2.html</a></summary>


|                          Subtest                           | Results | Messages |
|------------------------------------------------------------|---------|----------|
|                                                            | OK      |          |
| `integration test on reset for a created-from-script form` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-button-element/button-click-submits.html">/html/semantics/forms/the-button-element/button-click-submits.html</a></summary>


|                                      Subtest                                      |             Results              |     Messages     |
|-----------------------------------------------------------------------------------|----------------------------------|------------------|
|                                                                                   | TIMEOUT                          |                  |
| `clicking a button with .click() should trigger a submit`                         | **PASS: 5/10, MISSING: 5/10**    |                  |
| `clicking a button by dispatching an event should trigger a submit`               | **PASS: 5/10, MISSING: 5/10**    |                  |
| `clicking a button that prevents the event's default should not trigger a submit` | **TIMEOUT: 5/10, MISSING: 5/10** | `Test timed out` |
| `clicking a disabled button (via disabled attribute) should not trigger submit`   | **TIMEOUT: 5/10, MISSING: 5/10** | `Test timed out` |
| `clicking a disabled button (via ancestor fieldset) should not trigger submit`    | **TIMEOUT: 5/10, MISSING: 5/10** | `Test timed out` |
| `clicking a button inside a disabled fieldset's legend *should* trigger submit`   | **PASS: 5/10, MISSING: 5/10**    |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-button-element/button-type.html">/html/semantics/forms/the-button-element/button-type.html</a></summary>


|                            Subtest                             | Results | Messages |
|----------------------------------------------------------------|---------|----------|
|                                                                | OK      |          |
| `a button's type should be submit by default`                  | PASS    |          |
| `a button's type should stay within the range of valid values` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-input-element/checkbox-click-events.html">/html/semantics/forms/the-input-element/checkbox-click-events.html</a></summary>


|                                                       Subtest                                                       | Results |                         Messages                        |
|---------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------|
|                                                                                                                     | OK      |                                                         |
| `clicking and preventDefaulting a checkbox causes the checkbox to be checked during the click handler but reverted` | PASS    |                                                         |
| `a checkbox input emits click, input, change events in order after synthetic click`                                 | FAIL    | `assert_array_equals: lengths differ, expected 3 got 2` |
| `a checkbox input emits click, input, change events in order after dispatching click event`                         | FAIL    | `assert_array_equals: lengths differ, expected 3 got 2` |
| `checkbox input respects cancel behavior on synthetic clicks`                                                       | FAIL    | `assert_array_equals: lengths differ, expected 1 got 2` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-input-element/radio-input-cancel.html">/html/semantics/forms/the-input-element/radio-input-cancel.html</a></summary>


|                   Subtest                   | Results |                 Messages                |
|---------------------------------------------|---------|-----------------------------------------|
|                                             | OK      |                                         |
| `radio input cancel behavior reverts state` | FAIL    | `assert_false: expected false got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-label-element/proxy-click-to-associated-element.html">/html/semantics/forms/the-label-element/proxy-click-to-associated-element.html</a></summary>


|                                       Subtest                                        | Results | Messages |
|--------------------------------------------------------------------------------------|---------|----------|
|                                                                                      | OK      |          |
| `label with for attribute should proxy click events to the associated element`       | PASS    |          |
| `label without for attribute should proxy click events to the first labelable child` | PASS    |          |
| `clicking a label that prevents the event's default should not proxy click events`   | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-ask-for-a-reset.html">/html/semantics/forms/the-option-element/option-ask-for-a-reset.html</a></summary>


|             Subtest              | Results |                                             Messages                                            |
|----------------------------------|---------|-------------------------------------------------------------------------------------------------|
|                                  | OK      |                                                                                                 |
| `option element ask-for-a-reset` | FAIL    | `Error: assert_equals: Setting defaultSelected should reset others expected false but got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-element-constructor.html">/html/semantics/forms/the-option-element/option-element-constructor.html</a></summary>


|                                                                Subtest                                                                 | Results |                 Messages                |
|----------------------------------------------------------------------------------------------------------------------------------------|---------|-----------------------------------------|
|                                                                                                                                        | OK      |                                         |
| `Option constructor with no arguments`                                                                                                 | FAIL    | `assert_false: expected false got true` |
| `Option constructor with falsy arguments`                                                                                              | PASS    |                                         |
| `Option constructor creates HTMLOptionElement with specified text and value`                                                           | PASS    |                                         |
| `Option constructor handles selectedness correctly when specified with defaultSelected only`                                           | PASS    |                                         |
| `Option constructor handles selectedness correctly, even when incongruous with defaultSelected`                                        | PASS    |                                         |
| `Option constructor treats undefined text and value correctly`                                                                         | FAIL    | `assert_false: expected false got true` |
| `Option constructor treats falsy selected and defaultSelected correctly`                                                               | PASS    |                                         |
| `Option constructor treats truthy selected and defaultSelected correctly`                                                              | PASS    |                                         |
| `Option constructor does not set dirtiness (so, manipulating the selected content attribute still updates the selected IDL attribute)` | PASS    |                                         |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-index.html">/html/semantics/forms/the-option-element/option-index.html</a></summary>


|                                Subtest                                 | Results | Messages |
|------------------------------------------------------------------------|---------|----------|
|                                                                        | OK      |          |
| `option index should work inside the document`                         | PASS    |          |
| `option index should always be 0 for options in datalists`             | PASS    |          |
| `option index should always be 0 for options with no container`        | PASS    |          |
| `option index should always be 0 for options not even in the document` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-select-element/select-multiple.html">/html/semantics/forms/the-select-element/select-multiple.html</a></summary>


|                         Subtest                         | Results | Messages |
|---------------------------------------------------------|---------|----------|
|                                                         | OK      |          |
| `multiple selected options exist, both set from markup` | PASS    |          |
| `multiple selected options exist, one set from script`  | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-select-element/select-selectedOptions.html">/html/semantics/forms/the-select-element/select-selectedOptions.html</a></summary>


|                                                     Subtest                                                      | Results | Messages |
|------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                  | OK      |          |
| `.selectedOptions with no selected option`                                                                       | PASS    |          |
| `.selectedOptions with one selected option`                                                                      | PASS    |          |
| `.selectedOptions using the 'multiple' attribute with no selected options`                                       | PASS    |          |
| `.selectedOptions using the 'multiple' attribute with two selected options`                                      | PASS    |          |
| `.selectedOptions without the 'multiple' attribute but more than one selected option should return the last one` | PASS    |          |
| `.selectedOptions should always return the same value - [SameObject]`                                            | PASS    |          |
| `.selectedOptions should return the same object after selection changes - [SameObject]`                          | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html</a></summary>


|                                            Subtest                                             | Results | Messages |
|------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                | OK      |          |
| `defaultValue and value are the empty string by default`                                       | PASS    |          |
| `defaultValue and value are affected by setting textContent`                                   | PASS    |          |
| `defaultValue and value are affected by textContent in combination with appending a text node` | PASS    |          |
| `defaultValue and value treat CRLF differently`                                                | PASS    |          |
| `tests for the value setter`                                                                   | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/232599170)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 4dda02708fdfd837406b612708668c22b3c6d0b3
Using browser at version 14.14393
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>12 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/resetting-a-form/reset-form-2.html">/html/semantics/forms/resetting-a-form/reset-form-2.html</a></summary>


|                          Subtest                           | Results |                                          Messages                                          |
|------------------------------------------------------------|---------|--------------------------------------------------------------------------------------------|
|                                                            | OK      |                                                                                            |
| `integration test on reset for a created-from-script form` | FAIL    | `assert_equals: input should reset dirty checkedness to false expected false but got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-button-element/button-click-submits.html">/html/semantics/forms/the-button-element/button-click-submits.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-button-element/button-type.html">/html/semantics/forms/the-button-element/button-type.html</a></summary>


|                            Subtest                             | Results | Messages |
|----------------------------------------------------------------|---------|----------|
|                                                                | OK      |          |
| `a button's type should be submit by default`                  | PASS    |          |
| `a button's type should stay within the range of valid values` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-input-element/checkbox-click-events.html">/html/semantics/forms/the-input-element/checkbox-click-events.html</a></summary>


|                                                       Subtest                                                       | Results |                         Messages                        |
|---------------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------|
|                                                                                                                     | OK      |                                                         |
| `clicking and preventDefaulting a checkbox causes the checkbox to be checked during the click handler but reverted` | FAIL    | `assert_array_equals: lengths differ, expected 3 got 1` |
| `a checkbox input emits click, input, change events in order after synthetic click`                                 | FAIL    | `assert_array_equals: lengths differ, expected 3 got 2` |
| `a checkbox input emits click, input, change events in order after dispatching click event`                         | FAIL    | `assert_array_equals: lengths differ, expected 3 got 2` |
| `checkbox input respects cancel behavior on synthetic clicks`                                                       | FAIL    | `assert_array_equals: lengths differ, expected 1 got 2` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-input-element/radio-input-cancel.html">/html/semantics/forms/the-input-element/radio-input-cancel.html</a></summary>


|                   Subtest                   | Results |                         Messages                        |
|---------------------------------------------|---------|---------------------------------------------------------|
|                                             | OK      |                                                         |
| `radio input cancel behavior reverts state` | FAIL    | `assert_array_equals: lengths differ, expected 1 got 2` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-label-element/proxy-click-to-associated-element.html">/html/semantics/forms/the-label-element/proxy-click-to-associated-element.html</a></summary>


|                                       Subtest                                        | Results | Messages |
|--------------------------------------------------------------------------------------|---------|----------|
|                                                                                      | OK      |          |
| `label with for attribute should proxy click events to the associated element`       | PASS    |          |
| `label without for attribute should proxy click events to the first labelable child` | PASS    |          |
| `clicking a label that prevents the event's default should not proxy click events`   | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-ask-for-a-reset.html">/html/semantics/forms/the-option-element/option-ask-for-a-reset.html</a></summary>


|             Subtest              | Results | Messages |
|----------------------------------|---------|----------|
|                                  | OK      |          |
| `option element ask-for-a-reset` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-element-constructor.html">/html/semantics/forms/the-option-element/option-element-constructor.html</a></summary>


|                                                                Subtest                                                                 | Results |                 Messages                |
|----------------------------------------------------------------------------------------------------------------------------------------|---------|-----------------------------------------|
|                                                                                                                                        | OK      |                                         |
| `Option constructor with no arguments`                                                                                                 | PASS    |                                         |
| `Option constructor with falsy arguments`                                                                                              | PASS    |                                         |
| `Option constructor creates HTMLOptionElement with specified text and value`                                                           | PASS    |                                         |
| `Option constructor handles selectedness correctly when specified with defaultSelected only`                                           | PASS    |                                         |
| `Option constructor handles selectedness correctly, even when incongruous with defaultSelected`                                        | PASS    |                                         |
| `Option constructor treats undefined text and value correctly`                                                                         | PASS    |                                         |
| `Option constructor treats falsy selected and defaultSelected correctly`                                                               | PASS    |                                         |
| `Option constructor treats truthy selected and defaultSelected correctly`                                                              | FAIL    | `Type mismatch.\r\n`                    |
| `Option constructor does not set dirtiness (so, manipulating the selected content attribute still updates the selected IDL attribute)` | FAIL    | `assert_false: expected false got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-option-element/option-index.html">/html/semantics/forms/the-option-element/option-index.html</a></summary>


|                                Subtest                                 | Results | Messages |
|------------------------------------------------------------------------|---------|----------|
|                                                                        | OK      |          |
| `option index should work inside the document`                         | PASS    |          |
| `option index should always be 0 for options in datalists`             | PASS    |          |
| `option index should always be 0 for options with no container`        | PASS    |          |
| `option index should always be 0 for options not even in the document` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-select-element/select-multiple.html">/html/semantics/forms/the-select-element/select-multiple.html</a></summary>


|                         Subtest                         | Results | Messages |
|---------------------------------------------------------|---------|----------|
|                                                         | OK      |          |
| `multiple selected options exist, both set from markup` | PASS    |          |
| `multiple selected options exist, one set from script`  | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-select-element/select-selectedOptions.html">/html/semantics/forms/the-select-element/select-selectedOptions.html</a></summary>


|                                                     Subtest                                                      | Results | Messages |
|------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                  | OK      |          |
| `.selectedOptions with no selected option`                                                                       | PASS    |          |
| `.selectedOptions with one selected option`                                                                      | PASS    |          |
| `.selectedOptions using the 'multiple' attribute with no selected options`                                       | PASS    |          |
| `.selectedOptions using the 'multiple' attribute with two selected options`                                      | PASS    |          |
| `.selectedOptions without the 'multiple' attribute but more than one selected option should return the last one` | PASS    |          |
| `.selectedOptions should always return the same value - [SameObject]`                                            | PASS    |          |
| `.selectedOptions should return the same object after selection changes - [SameObject]`                          | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5878/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html</a></summary>


|                                            Subtest                                             | Results |                                                   Messages                                                  |
|------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------|
|                                                                                                | OK      |                                                                                                             |
| `defaultValue and value are the empty string by default`                                       | PASS    |                                                                                                             |
| `defaultValue and value are affected by setting textContent`                                   | FAIL    | `assert_equals: the defaultValue should reflect the textContent expected "foo bar" but got ""`              |
| `defaultValue and value are affected by textContent in combination with appending a text node` | FAIL    | `assert_equals: the defaultValue should reflect the textContent expected "foo bar baz" but got ""`          |
| `defaultValue and value treat CRLF differently`                                                | FAIL    | `assert_equals: the defaultValue should reflect the textContent expected "foo\r\nbar\rbaz\nqux" but got ""` |
| `tests for the value setter`                                                                   | FAIL    | `assert_equals: setting the value property should not affect the defaultValue expected "foo" but got ""`    |

</details>
</details>

**Review by tkent-google — APPROVED** [review]
lgtm

**Comment by domenic** [comment]
Bugs filed!!

- Gecko
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1365395
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1365396
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1365398
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1365399
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1365402
- Blink
  - https://bugs.chromium.org/p/chromium/issues/detail?id=723003
  - https://bugs.chromium.org/p/chromium/issues/detail?id=723004
  - https://bugs.chromium.org/p/chromium/issues/detail?id=534245
  - https://bugs.chromium.org/p/chromium/issues/detail?id=723006
- Edge
  - https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/12040569/
  - https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/12040708/
  - https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/12040790/
  - https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/12040797/
  - https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/12040806/
- WebKit
  - https://bugs.webkit.org/show_bug.cgi?id=172184
  - https://bugs.webkit.org/show_bug.cgi?id=149398
  - https://bugs.webkit.org/show_bug.cgi?id=172185

**Comment by cvrebert** [comment]
Seems these are redundant to each other:
* https://github.com/w3c/web-platform-tests/blob/891e76d9b5e1800bfbe9aaae1fa822940260ece4/html/semantics/forms/the-input-element/checkbox.html#L26
* https://github.com/w3c/web-platform-tests/blob/891e76d9b5e1800bfbe9aaae1fa822940260ece4/html/semantics/forms/the-input-element/checkbox-click-events.html#L57

**Pull request merged**